src/ApplicationInsights.Kubernetes/K8sEnvironmentFactory.cs

```diff
@@ -107,28 +109,26 @@ private async Task<V1Pod> SpinWaitUntilGetPodAsync(DateTime timeoutAt, Cancellat
                 // Try invoke a probe on readiness every 500ms until the container is ready
                 // Or it will timeout per the timeout settings.
                 await Task.Delay(500).ConfigureAwait(false);
-            } while (DateTime.Now < timeoutAt);
-
-            _logger.LogDebug($"{nameof(SpinWaitUntilGetPodAsync)} timed out.");
-            throw new InvalidOperationException("Can't find pod information in given time.");
+            } while (true);
         }
 
         /// <summary>
         /// Waits until the container is ready.
         /// Refer document @ https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase for the Pod's lifecycle.
         /// </summary>
-        /// <param name="timeoutAt">A point in time when the wait on container startup is abandoned.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>
         /// Returns the pod with refreshed info on it.
         /// </returns>
-        private async Task<V1ContainerStatus?> SpinWaitContainerReadyAsync(DateTime timeoutAt, CancellationToken cancellationToken)
+        private async Task<V1ContainerStatus?> SpinWaitContainerReadyAsync(CancellationToken cancellationToken)
         {
             Stopwatch stopwatch = new Stopwatch();
```

Comment:
This Stopwatch does not seep to be used anywhere?

Reply:
Probably from legacy code - I'll double-check.

Reply:
Removed.